src/paths.cpp

```diff
@@ -404,5 +404,352 @@ The underlying word graph. This function returns underlying
 :rtype:
    WordGraph
 )pbdoc");
+
+    py::options options;
+    options.disable_enum_members_docstring();
+
+    py::enum_<paths::algorithm>(m, "paths_algorithm", R"pbdoc(
+    An enum for specifying the algorithm to the functions :any:`number_of_paths`.
+
+The valid values are:
+
+.. py:attribute:: algorithm.dfs
+  :value: <algorithm.dfs: 0>
+
+  Use a depth-first-search.
+
+.. py:attribute:: algorithm.matrix
+  :value: <algorithm.matrix: 1>
+
+  Use the adjacency matrix and matrix multiplication.
+
+.. py:attribute:: algorithm.acyclic
+  :value: <algorithm.acyclic: 2>
+
+  Use a dynamic programming approach for acyclic word graphs.
+
+.. py:attribute:: algorithm.trivial
+  :value: <algorithm.trivial: 3>
+
+  Try to utilise some corner cases.
+
+.. py:attribute:: algorithm.automatic
+  :value: <algorithm.automatic: 4>
+
+  The function :any:`number_of_paths` tries to decide which algorithm is best.
+)pbdoc")
+        .value("dfs", paths::algorithm::dfs)
+        .value("matrix", paths::algorithm::matrix)
+        .value("acyclic", paths::algorithm::acyclic)
+        .value("trivial", paths::algorithm::trivial)
+        .value("automatic", paths::algorithm::automatic);
+
+    ////////////////////////////////////////////////////////////////////////
+    // Helper functions
+    ////////////////////////////////////////////////////////////////////////
+
+    m.def(
+        "paths_number_of_paths_algorithm",
+        [](WordGraph<uint32_t> const& wg, uint32_t source) {
+          return number_of_paths_algorithm(wg, source);
+        },
```

Comment:
For consistency with the rest of this file, I think it would be better (albeit equivalent) to use `node_type` instead of `uint32_t` here and in the following functions.